packages/oracle-manager/contracts/modules/core/OracleManagerModule.sol

```diff
@@ -0,0 +1,120 @@
+//SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "../../interfaces/IOracleManagerModule.sol";
+import "../../utils/ReducerNodeLibrary.sol";
+import "../../utils/ExternalNodeLibrary.sol";
+import "../../utils/PythNodeLibrary.sol";
+import "../../utils/ChainlinkNodeLibrary.sol";
+
+import "../../storage/Node.sol";
+import "../../storage/NodeDefinition.sol";
+
+contract OracleManagerModule is IOracleManagerModule {
+    error UnsupportedNodeType(uint nodeType);
+    error NodeNotRegistered(bytes32 nodeId);
+
+    function registerNode(
+        bytes32[] memory parents,
+        NodeDefinition.NodeType nodeType,
+        bytes memory parameters
+    ) external returns (bytes32) {
+        NodeDefinition.Data memory nodeDefinition = NodeDefinition.Data({
+            parents: parents,
+            nodeType: nodeType,
+            parameters: parameters
+        });
+
+        return _registerNode(nodeDefinition);
+    }
+
+    function getNodeId(
+        bytes32[] memory parents,
+        NodeDefinition.NodeType nodeType,
+        bytes memory parameters
+    ) external pure returns (bytes32) {
+        NodeDefinition.Data memory nodeDefinition = NodeDefinition.Data({
+            parents: parents,
+            nodeType: nodeType,
+            parameters: parameters
+        });
+
+        return _getNodeId(nodeDefinition);
+    }
+
+    function getNode(bytes32 nodeId) external pure returns (NodeDefinition.Data memory) {
+        return _getNode(nodeId);
+    }
+
+    function process(bytes32 nodeId) external view returns (Node.Data memory) {
+        return _process(nodeId);
+    }
+
+    function _getNode(bytes32 nodeId) internal pure returns (NodeDefinition.Data storage) {
+        return NodeDefinition.load(nodeId);
+    }
+
+    modifier onlyValidNodeType(NodeDefinition.NodeType nodeType) {
+        if (!_validateNodeType(nodeType)) {
+            revert UnsupportedNodeType(uint(nodeType));
+        }
+
+        _;
+    }
+
+    function _validateNodeType(NodeDefinition.NodeType nodeType) internal pure returns (bool) {
+        if (
+            NodeDefinition.NodeType.REDUCER == nodeType ||
+            NodeDefinition.NodeType.EXTERNAL == nodeType ||
+            NodeDefinition.NodeType.CHAINLINK == nodeType ||
+            NodeDefinition.NodeType.PYTH == nodeType
+        ) return true;
+
+        return false;
+    }
+
+    function _getNodeId(NodeDefinition.Data memory nodeDefinition) internal pure returns (bytes32) {
+        return NodeDefinition.getId(nodeDefinition);
+    }
+
+    function _registerNode(NodeDefinition.Data memory nodeDefinition)
+        internal
+        onlyValidNodeType(nodeDefinition.nodeType)
```

Comment:
i'd suggest moving this modifier to `registerNode` and passing in the `NodeType` that's passed in.